crates/cli-util/src/ui/mod.rs

```diff
@@ -8,22 +8,49 @@
 // the Business Source License, use of this software will be governed
 // by the Apache License, Version 2.0.
 
-use chrono::Duration;
-use chrono_humanize::{Accuracy, Tense};
+use chrono::{DateTime, TimeDelta};
+pub use chrono_humanize::{Accuracy, Tense};
+
+use crate::CliContext;
 
 pub mod console;
 pub mod output;
 pub mod stylesheet;
 pub mod watcher;
 
-pub fn duration_to_human_precise(duration: Duration, tense: Tense) -> String {
-    let duration = chrono_humanize::HumanTime::from(
-        Duration::try_milliseconds(duration.num_milliseconds()).expect("valid milliseconds"),
-    );
-    duration.to_text_en(Accuracy::Precise, tense)
+pub fn timestamp_as_human_duration<Tz: chrono::TimeZone>(ts: DateTime<Tz>, tense: Tense) -> String {
+    match CliContext::get().time_format() {
+        crate::opts::TimeFormat::Human => {
+            let since = ts.signed_duration_since(chrono::Local::now());
+            duration_to_human_precise(since, tense)
+        }
+        crate::opts::TimeFormat::Iso8601 => ts.to_rfc3339(),
+        crate::opts::TimeFormat::Rfc2822 => ts.to_rfc2822(),
```

Comment:
Aren't those timestamps and not durations as the function name suggests?

Reply:
I couldn't find a better name :) This will print the timestamp, but will calculate the elapsed duration only if the format is human. It's a little tricky and might be confusing but I don't have better ideas at the minute.